jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/LayoutModel.java

```diff
@@ -11,20 +11,40 @@
 
 import com.google.common.graph.Graph;
 import edu.uci.ics.jung.layout.algorithms.LayoutAlgorithm;
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
 import java.util.function.Function;
 
 /** two or three dimensional layoutmodel */
 public interface LayoutModel<N> extends Function<N, Point> {
 
+  /** @return the width of the layout area */
   int getWidth();
 
+  /** @return the height of the layout area */
   int getHeight();
 
+  /**
+   * allow the passed LayoutAlgorithm to operate on this LayoutModel
+   *
+   * @param layoutAlgorithm the algorithm to apply to this model's Points
+   */
   void accept(LayoutAlgorithm<N> layoutAlgorithm);
 
+  /**
+   * return a mapping of Nodes to Point locations
+   *
+   * @return
+   */
+  Map<N, Point> getLocations();
```

Comment:
I'm glad to see that you were able to bring this change back; what did you have to alter elsewhere in order to make it work?

Reply:
I'm thinking that I should not expose that method in the LayoutModel interface. To expose it might imply that the returned Map is in sync with the Graph. Maybe it is more of an implementation detail for the LoadingCacheLayoutModel. Having both the Graph::nodes() and the Map::keys() available breaks the single source of truth paradigm.

Reply:
updated, i believe your approach makes this work.